kafka/consumer_test.go

```diff
@@ -429,3 +431,277 @@ func TestConsumerLog(t *testing.T) {
+// This test uses 3 consumers while each consumer joins after the assignment
+// finished for the previous consumers.
+// The expected behavior for these consumers are:
+// 1) First consumer joins, AssignedPartitions happens. Assign all the
+//    partitions to it.
+// 2) Second consumer joins, RevokedPartitions happens from the first consumer,
+//    then AssignedPartitions happens to both consumers.
+// 3) Third consumer joins, RevokedPartitions happens from the previous two
+//    consumers, then AssignedPartitions happens to all the three consumers.
+// 4) Close the second consumer, RevokedPartitions should not happen.
+// 5) Rejoin the second consumer, rebalance should not happen to all the other
+//    consumers since it's not the leader, AssignedPartitions only happened
+//    to this consumer to assign the partitions.
+//
+// The total number of AssignedPartitions for the first consumer is 3,
+// and the total number of RevokedPartitions for the first consumer is 2.
+// The total number of AssignedPartitions for the second consumer is 2,
+// and the total number of RevokedPartitions for the second consumer is 1.
+// The total number of AssignedPartitions for the third consumer is 1,
+// and the total number of RevokedPartitions for the second consumer is 0.
+// The total number of AssignedPartitions for the rejoined consumer
+// (originally second consumer) is 1,
+// and the total number of RevokedPartitions for the rejoined consumer
+// (originally second consumer) is 0.
+func TestConsumerCloseForStaticMember(t *testing.T) {
+	if !testconfRead() {
+		t.Skipf("Missing testconf.json")
+	}
+	broker := testconf.Brokers
+	topic := testconf.Topic
+
+	var assignedEvents1 int32
+	var revokedEvents1 int32
+
+	var assignedEvents2 int32
+	var revokedEvents2 int32
+
+	var assignedEvents3 int32
+	var revokedEvents3 int32
+
+	var assignedEvents4 int32
+	var revokedEvents4 int32
+
+	conf1 := ConfigMap{
+		"bootstrap.servers":    broker,
+		"group.id":             "rebalance",
+		"session.timeout.ms":   "6000",
+		"max.poll.interval.ms": "10000",
+		"group.instance.id":    "staticmember1",
+	}
+	c1, err := NewConsumer(&conf1)
+
+	conf2 := ConfigMap{
+		"bootstrap.servers":    broker,
+		"group.id":             "rebalance",
+		"session.timeout.ms":   "6000",
+		"max.poll.interval.ms": "10000",
+		"group.instance.id":    "staticmember2",
+	}
+	c2, err := NewConsumer(&conf2)
+	if err != nil {
+		t.Fatalf("%s", err)
+	}
+
+	conf3 := ConfigMap{
+		"bootstrap.servers":    broker,
+		"group.id":             "rebalance",
+		"session.timeout.ms":   "6000",
+		"max.poll.interval.ms": "10000",
+		"group.instance.id":    "staticmember3",
+	}
+
+	c3, err := NewConsumer(&conf3)
+	if err != nil {
+		t.Fatalf("%s", err)
+	}
+	wrapRebalancecb1 := wrapRebalanceCb(&assignedEvents1, &revokedEvents1, t)
+	err = c1.Subscribe(topic, wrapRebalancecb1)
+	if err != nil {
+		t.Fatalf("Failed to subscribe to topic %s: %s\n", topic, err)
+		return
+	}
+
+	wg := &sync.WaitGroup{}
```

Comment:
I think it is better to keep this as a non-pointer here, and then pass it as pointer when needed (to testPoll()).

Reply:
If we keep this as a non-pointer here, we need to use it as pointer each time, and there are a lot of use cases in the same function. 

For example:
wg.Add(1)
wg.Wait()

Reply:
The pointer receiver methods apply also to non pointer variables, based on [this](https://go.dev/tour/methods/6).

So pointer should be needed to pass the WaitGroup struct by reference, but not to call its methods with pointer receiver.

Reply:
> If we keep this as a non-pointer here, we need to use it as pointer each time, and there are a lot of use cases in the same function.

I still believe it is more correct to keep it as a non-pointer here and pass it as a pointer where needed.
But it is a nit, so do as you wish.